jujugui/static/gui/src/app/components/deployment-flow/sshkey/sshkey.js

```diff
@@ -30,15 +30,310 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 const AZURE_CLOUD_TYPE = 'azure';
 
 class DeploymentSSHKey extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      addSource: 'github',
+      SSHkeys: [],
+      error: null,
+      buttonDisabled: true
+    };
+  }
+
+  /** Handle keyboard input to listen for return.
+
+    @method _onSSHKeyInputKey
+    @param {Object} evt The keyboard event.
+  */
+  _onSSHKeyInputKey(evt) {
+    if (evt.which === 13) {
+      this._handleAddMoreKeys(null);
+    }
+    this.setState({
+      buttonDisabled: this.refs.sshKey.getValue() ? false : true
+    });
+  }
+
+  /** Handle keyboard input to listen for return.
+
+    @method _onSSHKeyInputKey
+    @param {Object} evt The keyboard event.
+  */
+  _onGithubUsernameInputKey(evt) {
+    if (evt.which === 13) {
+      this._handleAddMoreKeys(null);
+    }
+    this.setState({
+      buttonDisabled: this.refs.githubUsername.getValue() ? false : true
+    });
```

Comment:
Those two methods are very similar. Maybe we can use a single method receiving the ref name and an evt, and then bind the ref name in the caller?